### ISSUE TITLE
DataBinding for two layouts with same attributes. Algolia instant search for Android

### ISSUE THREAD

**Issue by dmkdfc** [algolia/instantsearch-android#60]
I have created two activities whose xml have same attributes. When i use instant search the results displayed are not getting removed instead same display page is shown in other xml. Stucked since 9 long hours !! plz help

**Comment by PLNech** [comment]
Hi @dmkdfc, thanks for your report!

I don't get what you mean by `whose xml have same attributes`. Can you share here the two layouts?

**Comment by dmkdfc** [comment]
The thing i am asking is can i use 2 different algolia app ids in my single
android app !? If so then issue what i am encountering is the data which is
showed by hits widget it remains the same throught though i use 2 app ids
with different data set in algolia !

**Comment by PLNech** [comment]
We never tried using two different applications in the same InstantSearch Android interface, but you can use two different `Searcher` each created with its own Application id and API Key. However you will need to manually link each one to its widgets, so it's not a very easy way to build an interface.

Can you tell me more about your use case? There is likely an easier way to achieve what you are trying to build.

**Comment by dmkdfc** [comment]
See basically i have two sections !
1) Search in teachers news
2) Search in students news

I have used 2 different algolia apps for this scenario as both above
mentioned thing have different set of databases.

If i first search using algolia search for teachers news and then go back
to students news again i get the same data which was opened first i.e here
i opened teachers news first i searched in that went back to students news
now students news data should be loaded ! But what is happening is whatever
you open first same gets repeted after that.

**Comment by PLNech** [comment]
Thanks, I understand better now. 

What you are looking for is *multi-index search* capabilities. This will let you target two different indices at the same time, like in the following example:

![multi-index](https://user-images.githubusercontent.com/1821404/36109152-e5d7525a-101e-11e8-8b51-e9895af46189.gif)

This feature is still under development, but will be released soon (see PR #59). I'll keep you updated here as soon as it is available!

_One side note: you don't need [several Applications](https://www.algolia.com/manage/applications), only [several indices](https://www.algolia.com/explorer/indices), to implement multi-index search. It's way easier to manage several indices in the same app than to switch between applications._

**Comment by dmkdfc** [comment]
Exactly ! I saw this in your github under movie example but later it shows
404 not found. But instead of using two indices i want to use 2 algolia
apps in my single android app. I think dataBinding is causing the problem
whatever data gets binded the first time it doesnt get changed. I tried
giving different ids to views then it throws an error getClass() object
pointing to null pointer exception.

**Comment by PLNech** [comment]
Hi @dmkdfc, we just released [v1.5.0](https://github.com/algolia/instantsearch-android/releases/tag/1.5.0) which ships multi-index capabilities.

You can use it with two indices in the same app, or with two different apps:
```java
// Instantiate your Searchers
searcherMovies = Searcher.create(ALGOLIA_APP_ID, ALGOLIA_API_KEY, ALGOLIA_INDEX_MOVIES);
searcherActors = Searcher.create(ALGOLIA_APP_ID, ALGOLIA_API_KEY, ALGOLIA_INDEX_ACTORS);
/* or with several apps:
searcherActors = Searcher.create(ALGOLIA_OTHER_APP_ID, ALGOLIA_OTHER_API_KEY, ALGOLIA_INDEX_ACTORS);
*/
// Create a SearchBoxViewModel with your SearchBox
searchBoxViewModel = new SearchBoxViewModel(this.findViewById(R.id.include_searchbox););
// Link it to your search widgets
new InstantSearch(this, searcherMovies).registerSearchView(this, searchBoxViewModel);
new InstantSearch(this, searcherActors).registerSearchView(this, searchBoxViewModel);
```

The Hit's [databinding](https://community.algolia.com/instantsearch-android/widgets.html#data-binding) will bind each widget to the appropriate searcher, you can select which widget is for each searcher by setting the `algolia:variant` to the indexName:

```xml
<TextView
    android:id="@+id/actorName"
    android:layout_width="wrap_content"
    android:layout_height="wrap_content"
    algolia:attribute='@{"name"}'
    algolia:variant='@{"actors"}'
/>

<TextView
    android:id="@+id/movieName"
    android:layout_width="wrap_content"
    android:layout_height="wrap_content"
    algolia:attribute='@{"name"}'
    algolia:variant='@{"movies"}'
/>
```

Let me know if this works for you. If you have any problem with that solution, please reopen this issue 🙂

**Comment by dmkdfc** [comment]
Will work on it and get back to you soon !!

On Mar 2, 2018 8:35 PM, "Paul-Louis NECH" <notifications@github.com> wrote:

Closed #60 <https://github.com/algolia/instantsearch-android/issues/60>.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/algolia/instantsearch-android/issues/60#event-1501322825>,
or mute the thread
